src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -78535,6 +78535,61 @@ static class Extensions
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void ForEach_ExtensionGetEnumerator16()
+        {
+            var source = @"
+using System.Collections.Generic;
+
+#nullable enable
+public class C<T> {
+    static void M(C<string> c)
+    {
+        foreach (var i in c)
+        {
+        }
+    }
+}
+#nullable disable
+public static class CExt
+{
+    public static IEnumerator<int> GetEnumerator<T>(this C<T> c, T t = default) => throw null;
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,27): warning CS8620: Argument of type 'C<string>' cannot be used for parameter 'c' of type 'C<string?>' in 'IEnumerator<int> CExt.GetEnumerator<string?>(C<string?> c, string? t = null)' due to differences in the nullability of reference types.
```

Comment:
>C<string?> [](start = 114, length = 10)

I'm not following something. Where does the `C<string?>` come from?

Reply:
Because the `T t = default`. `string` comes from the first argument, and the second argument changes that to nullable because it must be.

---
In reply to: [549493723](https://github.com/dotnet/roslyn/pull/50105#discussion_r549493723) [](ancestors = 549493723)